File: learn-java/src/test/java/boki/learnjava/stepverifier/publisherprobe/PublisherProbeTestEx1.java
package boki.learnjava.stepverifier.publisherprobe;


import boki.learnjava.stepverifier.PublisherProbeExample;
import org.junit.jupiter.api.Test;
import reactor.test.StepVerifier;
import reactor.test.publisher.PublisherProbe;

class PublisherProbeTestEx1 {

    @Test
    void publisherProbeTest() {
        PublisherProbe<String> probe = PublisherProbe.of(PublisherProbeExample.useStandbyPower());

        StepVerifier
            .create(PublisherProbeExample.processWith(PublisherProbeExample.useMainPower().log(), probe.mono().log()))
            .expectNextCount(1)
            .verifyComplete();

        probe.assertWasSubscribed();
        probe.assertWasRequested();
        probe.assertWasNotCancelled();
    }
}

File: learn-kt/src/test/kotlin/boki/learn/stepverifier/publisherprobe/PublisherProbeTestEx1.kt
package boki.learn.stepverifier.publisherprobe

import boki.learn.stepverifier.PublisherProbeExample
import org.junit.jupiter.api.Test
import reactor.test.StepVerifier
import reactor.test.publisher.PublisherProbe

class PublisherProbeTestEx1 {

    @Test
    fun publisherProbeTest() {
        val probe = PublisherProbe.of(PublisherProbeExample.useStandbyPower())

        StepVerifier
            .create(
                PublisherProbeExample.processWith(
                    PublisherProbeExample.useMainPower().log(), probe.mono().log()
                )
            )
            .expectNextCount(1)
            .verifyComplete()

        probe.assertWasSubscribed()
        probe.assertWasRequested()
        probe.assertWasNotCancelled()
    }
}